### PR TITLE
Security Advisories: Add information from Riscure mini-audit

### PR DESCRIPTION
**Pull request by jbech-linaro**
Out of all potentially security issues reported by Riscure as an outcome
of their "mini-audit" we narrowed down seven of them as higher severity.
These are the ones that has been mentioned in this patch.

CVE numbers have been requested, but we haven't got them yet. If/when we
get those, we will submit another update patch.

Signed-off-by: Joakim Bech <joakim.bech@linaro.org>

### CODE REVIEW

File: security-advisories/README.md [hunk 1]
```diff
@@ -14,9 +14,178 @@ At this page we will list of all known security vulnerabilities found on OP-TEE.
...
+not able to observe any data belonging to a TA previously stored on this exact
+location in memory, the memory block is set to zero. The size used to memset
+the block to zero is the sum of the sizes of the segments, not the rounded size
+of the actual allocation. This means that the remaining space at the end of the
+allocation is not cleared, potentially leaking code and/or data of a previous
+TA. The information gained by this attack is limited by the memory layout of
+the (compromised) TA performing the attack and the flags (i.e. is unloading the
+TA prevented when the last session is closed due to
+`TA_FLAG_INSTANCE_KEEP_ALIVE`) and layout of the attacked TA. Therefore we
+consider the risk to be
```
Comment (jforissier) [inline security-advisories/README.md]: ...to be what? (same below)

File: security-advisories/README.md [hunk 2]
```diff
@@ -14,9 +14,178 @@ At this page we will list of all known security vulnerabilities found on OP-TEE.
...
+However, the function `tee_mmu_check_access_rights` does not check every page of
+the TA provided buffer. A TA provides a buffer as a pointer (uaddr) and a
+length (a). The provided buffer is checked piecewise in increments of addr_incr
+(4KiB) in a for-loop. In the case where len is not already page aligned, the
+termination condition a < (uaddr + len) has been passed when addr_incr is added
+the last time of the loop iteration. Therefore, the final page of the TA
+provided buffer is not checked. A TA could provide a buffer of which up to 4KiB
+resides in the context of the TEE kernel or another TA. This could lead to
+memory corruption of the TEE itself or another TA. Memory corruption
+vulnerabilities can have
```
Comment (jforissier) [inline security-advisories/README.md]: ...can have what?

File: security-advisories/README.md (hunk 1 above)
Comment (jbech-linaro) [reply security-advisories/README.md]: Let it up for the reader to decide, that makes it more like a storytale :smile: 

No seriously, copy/paste bug from the Riscure doc, I'll fix on Monday.

File: security-advisories/README.md (hunk 2 above)
Comment (jbech-linaro) [reply security-advisories/README.md]: Copy/paste bug from the Riscure doc, I'll fix on Monday.

File: security-advisories/README.md (hunk 1 above)
Comment (jbech-linaro) [reply security-advisories/README.md]: It should end with "medium", but I think I', gonna remove that last sentence entirely. It doesn't really belong in the text at the security advisories page I think.

**Comment by jbech-linaro** [comment]
@jforissier , removed the sentences about severity and tweaked some other minor things, please have a look again.

**Comment by jforissier** [comment]
LGTM

**Comment by jbech-linaro** [comment]
@kylekirkby , I think this is ready to be merged. The link checker etc fails for other reason (we have removed *.md files).

**Comment by kylekirkby** [comment]
@jbech-linaro -  the link checker found these broken  links:

/agreement/index.html:
   https://github.com/OP-TEE/optee_os/blob/master/Notice.md#contributions [404]
   https://github.com/OP-TEE/optee_os/tree/master/documentation [404]
   https://github.com/OP-TEE/optee_os/blob/master/documentation/build_system.md [404]
   https://github.com/OP-TEE/optee_os/blob/master/documentation/optee_design.md [404]
/blog/op-tee-qa/index.html:
   https://github.com/OP-TEE/optee_os/blob/master/documentation/secure_storage.md [404]
   https://github.com/OP-TEE/optee_os/blob/master/documentation/secure_storage_rpmb.md [404]
/documentation/index.html:
   https://boundarydevices.com/product/sabre-lite-imx6-sbc/ [404]
/faq/index.html:
   https://github.com/OP-TEE/optee_os/blob/master/Notice.md [404]
   https://github.com/OP-TEE/optee_os/tree/master/documentation [404]
   https://github.com/OP-TEE/optee_os/blob/master/documentation/crypto.md [404]
   https://github.com/OP-TEE/optee_os/blob/master/documentation/porting_guidelines.md [404]
/usage/index.html:
   https://github.com/OP-TEE/optee_os/blob/master/documentation/github.md [404]

Are the equivalents for these pages all available in the readthedocs documentation?

**Comment by jbech-linaro** [comment]
> Are the equivalents for these pages all available in the readthedocs documentation?

Yes, there are. But those links are not related to this particular patch set? Right? I see that as a separate thing that needs to be fixed. Or?

**Comment by kylekirkby** [comment]
@jbech-linaro, I can fix them if I know where the new pages are located - if that helps ? Maybe we merge this knowing it will fail - and I can then create a new pull request with the fixes for the above ?

**Comment by jbech-linaro** [comment]
Since the link errors are unrelated to this particular patch, I think the right thing is to do this in two different patches. If the link errors would have been introduced by this patch, then it would have been another question.

**Comment by jbech-linaro** [comment]
Rebased and pushed again.

**Pull request merged**